include/toolchain/common.h

```diff
@@ -98,7 +98,8 @@
 /* Indicate that an array will be used for stack space. */
 
 #if !defined(_ASMLANGUAGE)
-  #define __stack __aligned(STACK_ALIGN)
+  /* don't use this anymore, use K_DECLARE_STACK instead. Remove for 1.11 */
```

Comment:
Do we have a macro for version checking? Might be useful for future deprecations:

```C
#if ZEPHYR_CHECK_VERSION(1,11)
#error Remove definition for __stack
#endif
``

(glib/gtk has a similar macro, and it's pretty useful.)